swiftpkg/internal/swiftpkg_build_files.bzl

```diff
@@ -106,6 +106,9 @@ def _swift_target_build_file(pkg_ctx, target):
         # SPM directive instructing the code to build as if a Swift package.
         # https://github.com/apple/swift-package-manager/blob/main/Documentation/Usage.md#packaging-legacy-code
         "-DSWIFT_PACKAGE",
+        # SPM directive instructing the code to build as if a Swift package for any clang modules.
+        "-Xcc",
+        "-DSWIFT_PACKAGE",
```

Comment:
I am confused. When are we building a C/C++ file with the swift compiler? We have separate targets for Swift and C/C++ source code.

Reply:
-Xcc passes a flag through to all C compiler invocations

Reply:
Sorry. Perhaps, I am missing something. We create `cc_xxx`, `objc_xxx`, and `swift_xxx` targets separating the source files into their respective targets. How is a C/C++ file being compiled by the Swift compiler? The `cc_xxx` and `objc_xxx` should be compiled with clang.

Reply:
When Swift depends on a clang module (modulemap and headers), with implicit modules it compiles that clang module. `-Xcc` is a way to pass compiler flags to that compilation.